addons/project_mail_plugin/controllers/project_client.py

```diff
@@ -0,0 +1,41 @@
+
+from odoo import http
+from odoo.http import request
+from odoo.tools import html2plaintext
+
+
+class ProjectClient(http.Controller):
+    @http.route('/mail_plugin/project/search', type='json', auth='outlook', cors="*")
+    def projects_search(self, search_term, limit=5):
+
+        projects = request.env['project.project'].search([('name', 'ilike', search_term)], limit=limit)
+
+        return [
+            {
+                'name': project.name,
+                'project_id': project.id,
+            }
+            for project in projects
+        ]
+
+    @http.route('/mail_plugin/task/create', type='json', auth='outlook', cors="*")
```

Comment:
Was wondering if we should enforce URLs to contain "project", since this is the mail_plugin project implementation, so you would have something like '/mail_plugin/project/task/create' here.

But I see this was not done in helpdesk either so keep it as is for now.

Reply:
Okay, we can keep it like this.